atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbReadTable.java

```diff
@@ -105,7 +105,7 @@ public DbReadTable(ConnectionSupplier conns, DbQueryFactory queryFactory) {
         FullQuery query = queryFactory.getRowsColumnRangeCountsQuery(rows, ts, columnRangeSelection);
         AgnosticLightResultSet results = conns.get()
                 .selectLightResultSetUnregisteredQuery(query.getQuery(), query.getArgs());
-        results.setFetchSize(Math.max(rows.size(), MAX_ROW_COLUMN_RANGES_FETCH_SIZE));
+        results.setFetchSize(Math.min(rows.size(), MAX_ROW_COLUMN_RANGES_FETCH_SIZE));
```

Comment:
wondering if we should bail quickly if rows is empty to avoid unnecessary work and fetch size 0.

```
if (rows.isEmpty()) return ClosableIterators#emptyImmutableClosableIterator();
```

Per https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html#setFetchSize-int-
>  If the fetch size specified is zero, the JDBC driver ignores the value and is free to make its own best guess as to what the fetch size should be.